apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/TestMessageHandler.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Storage.V1.Tests;
+
+internal class TestMessageHandler : ReplayingMessageHandler
```

Comment:
It's not clear to me that we need a new class here. I'd add the functionality to `ReplayingMessageHandler` - probably not making the assertions directly, but just recording the time of each request. Have a look at StorageClientImplRetryTest for an example of adding extra checking.